src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -104,33 +103,33 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                             // No packages means that no package references at all were found in the current framework
                             if (!packages.Any())
                             {
-                                Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoPackagesFoundForFrameworks, projectName));
+                                projectModel.AddProjectProblem(error: string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoPackagesFoundForFrameworks, projectModel.ProjectName), problemType: ProblemType.Information);
                             }
                             else
                             {
                                 if (listPackageArgs.ReportType != ReportType.Default)  // generic list package is offline -- no server lookups
                                 {
                                     PopulateSourceRepositoryCache(listPackageArgs);
-                                    WarnForHttpSources(listPackageArgs);
+                                    WarnForHttpSources(listPackageArgs, projectModel);
                                     await GetRegistrationMetadataAsync(packages, listPackageArgs);
                                     await AddLatestVersionsAsync(packages, listPackageArgs);
                                 }
 
-                                bool printPackages = FilterPackages(packages, listPackageArgs);
+                                bool printPackages = projectModel.PrintPackagesFlag;
 
                                 // Filter packages for dedicated reports, inform user if none
                                 if (listPackageArgs.ReportType != ReportType.Default && !printPackages)
                                 {
                                     switch (listPackageArgs.ReportType)
                                     {
                                         case ReportType.Outdated:
-                                            Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoUpdatesForProject, projectName));
+                                            projectModel.AddProjectProblem(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoUpdatesForProject, projectModel.ProjectName), ProblemType.Information);
```

Comment:
It's kind of weird that we're calling them problems, when they are mostly just messages :) 
Maybe we can have a better name?

Reply:
Rename to `AddProjectInformation`. If you have better suggestion, please let me know.

Reply:
Works for me. Resovling the thread.